### PR TITLE
Add imagemap support to bbcode

### PR DESCRIPTION
**Pull request by nanaya**
Syntax support for #8751. Frontend for it is missing.

I just hope enterprising user will create one if there's enough interest 👀

```
[imagemap]
imageurl
x y width height url1 optional title
x y width height url2 optional title
x y width height # hover title
[/imagemap]
```

All the units (x, y, width, height) are in percent without the percent sign. Passing `#` as url will make the area only show title on hover (or nothing if no title specified).

```
[imagemap]
https://assets.ppy.sh/osu-web-test-resources/placeholder-1280x720.jpg
0 0 50 50 https://google.com q1
0 50 50 50 https://yahoo.com q2
50 0 50 50 https://osu.ppy.sh q3
50 50 50 50 # q4
[/imagemap]
```

### CODE REVIEW

**Comment by RockRoller01** [comment]
Do I see it correctly that the URL is **not** optional? From my own experience with current imagemaps, making the url optional seems benefitial. For example in skinning we often only want some parts of the imagemap to be a link

**Comment by nanaya** [comment]
image url isn't optional otherwise there's nothing to map to...

**Comment by RockRoller01** [comment]
ofc, my comment was solely focused on the areas you can define for an imagemap

**Comment by nanaya** [comment]
Now I think again, I'm not sure if the positioning should use percent or pixel 🤔 if anyone have preference of one over the other please comment here.

**Comment by RockRoller01** [comment]
> if the positioning should use percent or pixel 

Looking at the code, wouldn't pixel result in a non-responsive design?

**Comment by nanaya** [comment]
It'll be converted to percent

File: app/Libraries/BBCodeFromDB.php [hunk 1]
```diff
@@ -129,6 +129,54 @@ public function parseHeading($text)
...
+                                    "width: {$link[2]}%",
+                                    "height: {$link[3]}%",
+                                ]),
+                                'title' => $link[5] ?? '',
+                            ]),
+                            $links,
+                        ));
+
+                        $imageUrl = proxy_media($map['imageUrl']);
+                        $imageSize = fast_imagesize($imageUrl) ?? [0, 0];
```
Comment (notbakaneko) [inline app/Libraries/BBCodeFromDB.php:162]: I think there needs to be something better than rendering a 0x0 image if it fails to get the size, or just skip rendering the image. It's also not apparent to the user if something goes wrong here, they just get an image that gets displayed at 0px

**Review by notbakaneko — CHANGES_REQUESTED** [review]

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**